src/screens/WebsocketResearchScreen/SignedChatScreen.tsx

```diff
@@ -22,13 +26,21 @@ const SignedChatScreen = () => {
     goBackFromChatScreen,
     goToChatInfoScreen,
     sendChat,
-    updateChatContinuity
+    updateChatContinuity,
+    flatListRef: scrollRef,
+    scrollContext
   } = useChatScreenHook(SIGNED);
+  const {replyPreview, clearReplyPreview} = useMessageHook();
+  const updatedChats = updateChatContinuity(chats);
+  const [loading, setLoading] = React.useState(false);
+  const {signedProfileId} = useProfileHook();
 
-  const flatlistRef = React.useRef<FlatList>();
   const [, dispatchChannel] = (React.useContext(Context) as unknown as any).channel;
   const [profile] = (React.useContext(Context) as unknown as any).profile;
-  const updatedChats = updateChatContinuity(chats);
+
+  const memberChat = selectedChannel?.rawJson?.channel?.members?.find(
+    (item: any) => item.user_id !== signedProfileId
+  );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `updateChatContinuity` function is now being called inside the component render. This could potentially lead to unnecessary re-renders if `chats` changes frequently. Consider using a `useMemo` hook to only recalculate `updatedChats` when `chats` changes.
```diff
- const updatedChats = updateChatContinuity(chats);
+ const updatedChats = React.useMemo(() => updateChatContinuity(chats), [chats]);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->